### PR TITLE
Blogit: - Feature/validations testing 🧪 ⚖️ 👨🏾‍💻 

### PR DESCRIPTION
**Pull request by Bennyjoez**
## Blogit: - Feature/validations testing 🧪 ⚖️ 👨🏾‍💻  🪶 

In this milestone, I managed to implement the following in my code: 📄 
- Add the following validations: ➕ 
   - For the User model: 👤 
   - Name must not be blank. 📛 
   - PostsCounter must be an integer greater than or equal to zero. ✔️ 
- For the Post model: 
   - Title must not be blank. ✔️ 
   - Title must not exceed 250 characters. ✔️ 
   - CommentsCounter must be an integer greater than or equal to zero. ✔️ 
   - LikesCounter must be an integer greater than or equal to zero. ✔️ 
- Add unit specs for all of your models' methods and validations. ✔️ 

💯

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@ class User < ApplicationRecord
   has_many :likes, foreign_key: 'author_id'
   has_many :comments, foreign_key: 'author_id'
 
+  validates :name, presence: true
```
Comment (VICTOR-CHUKWU) [inline app/models/user.rb:8]: - [x] kindly add validation for post_counter to be an integer greater than or equal to zero.

File: app/models/post.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@ class Post < ApplicationRecord
 
   after_create :update_user_posts_counter
 
+  validates :title, presence: true, length: { in: 3..250 }
```
Comment (VICTOR-CHUKWU) [inline app/models/post.rb:10]: - [x] kindly add validation for CommentsCounter must be an integer greater than or equal to zero. and LikesCounter must be an integer greater than or equal to zero.

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Bennyjoez  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 3]
```diff
@@ -5,6 +5,10 @@ class Post < ApplicationRecord
...
+  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, allow_nil: true
+  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, allow_nil: true
```
Comment (christianbiring1) [inline app/models/post.rb]: [OPTIONAL]
- [ ] I would suggest not accepting a `nil` value to your variables as they cause all sorts of bugs in your program, Kindly you may consider removing that. Rather you may consider initializing these variables (like, comment_counter) to zero

**Review by christianbiring1 — APPROVED** [review]
Hi @Bennyjoez , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 

## Highlights
- [ ] Great work on implementing the changes as suggested by the previous reviewer :+1: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 3 above)
Comment (Bennyjoez) [reply app/models/post.rb]: Thank you for the suggestions. I have implemented them. 👍🏾

**Pull request merged**